### DISCUSSION TITLE
How do I "merge cells" or make one section into sub sections?

### DISCUSSION THREAD

**Discussion by Piste** [Lissy93/dashy#971] [Help and Support] [answered]
I feel like this is super basic, so I hesitated before asking: how do I make sure my first long section does not force a weird look? Ideally, with 5 sections, I'd like the first section to take 2 rows. Even better, I'd like not to have to worry about this and just let Dashy automatically layout perfectly :p

<img width="1574" alt="Pasted_Image_11_6_22__11_29" src="https://user-images.githubusercontent.com/5099194/200182716-293dc771-fe89-4de1-b832-1036c4cb36a6.png">

**Comment by Piste** [comment, marked as answer]
And this is how:

<img width="821" alt="Pasted_Image_11_6_22__11_41" src="https://user-images.githubusercontent.com/5099194/200183188-af9be543-0f40-4653-b0d0-f4408f9f0042.png">

Funny how writing down a question forces you to use concepts and words that get you closer to a solution, hey...

**Reply by lissy93** [reply]
Nice, glad you figured it out :)

I must admit, the current method, isn't the most  intuitive, something I should improve upon in the future...

**Comment by lissy93** [comment]
This can be done by setting the `rows` and `cols` attributes, defined under [`displayData`](https://github.com/Lissy93/dashy/blob/master/docs/configuring.md#sectiondisplaydata-optional) of a given section. By default each is `1`, but increasing the number will make it span multiple rows, or columns,

---

- `cols` will specify the number of columns a given section should span horizontally, i.e width
   -   uses `grid-column-start: span [X]` under the hood
- `rows` will specify the number of rows a given section should span vertically, i.e height
  - uses `grid-row-start: span [X]` under the hood

<p align="center">
<img width="500" src="https://i.ibb.co/HXRWVRK/how-rows-and-cols-work-in-dashy.png" />
</p>

---

See also:
- https://github.com/Lissy93/dashy/discussions/148#discussioncomment-1449275
- https://github.com/Lissy93/dashy/issues/7
